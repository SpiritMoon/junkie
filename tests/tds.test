#!/bin/sh

NAME="TDS"
PCAPS="pcap/tds/ms-sql-tds-rpc-requests.pcap pcap/tds/mssql_truncated.pcap pcap/tds/mssql_with_accent.pcap"
filter() {
	grep '^TDS'
}
OUTPUT="tds.out"
CMDLINE='(use-modules (junkie signatures))'

. "$srcdir/dump_test"
