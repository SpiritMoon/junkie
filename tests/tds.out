-- pcap/tds/ms-sql-tds-rpc-requests.pcap --
TDS: head_len=8, payload=182, type=SQL batch, status=0x1
TDS(msg): head_len=182, payload=0, Clt->Srv, query, query=' set transaction isolation level  read committed  set implicit_transactions off '
TDS: head_len=8, payload=26, type=Tabular result, status=0x1
TDS(msg): head_len=26, payload=0, Srv->Clt, query, nb_rows=-1, nb_fields=-1, Status=Completed
TDS: head_len=8, payload=284, type=RPC, status=0x1
TDS(msg): head_len=284, payload=0, Clt->Srv, query, query='PrepExec(?*=0,?=@P0 nvarchar(4000),@P1 int,?=select * from test_table_1 where name = @P0 and id = @P1                ,?=zzz,?=2)'
TDS: head_len=8, payload=350, type=Tabular result, status=0x1
TDS(msg): head_len=350, payload=0, Srv->Clt, query, nb_rows=1, nb_fields=4, Status=Completed
TDS: head_len=8, payload=36, type=SQL batch, status=0x1
TDS(msg): head_len=36, payload=0, Clt->Srv, query, query='COMMIT TRANSACTION'
TDS: head_len=8, payload=9, type=Tabular result, status=0x1
TDS(msg): head_len=9, payload=0, Srv->Clt, query, nb_rows=-1, nb_fields=-1, Status=Completed
TDS: head_len=8, payload=177, type=RPC, status=0x9
TDS(msg): head_len=177, payload=0, Clt->Srv, query, query='p_GetBogusData(@SearchType=1,@MaxWaitTimeInSeconds=0,@ProcessNegativeAck=0)'
TDS: head_len=8, payload=1074, type=RPC, status=0x9
TDS(msg): head_len=1074, payload=0, Clt->Srv, query, query='sp_executesql(?=SELECT TOP 88 [dbo].[MyExampleTable].[ID], [dbo].[MyExampleTable].[EntityID], [dbo].[MyExampleTable].[EntityHistoryID], [dbo].[MyExampleTable].[EntityTypeID], [dbo].[MyExampleTable].[EntityVersion], [dbo].[MyExampleTable].[HandlingStatus], [dbo].[MyExampleTable].[OperationTypeID] FROM [dbo].[MyExampleTable]  WHERE ( ( ( [dbo].[MyExampleTable].[HandlingStatus] = @HandlingStatus1))) ORDER BY [dbo].[MyExampleTable].[EntityVersion] ASC,[dbo].[MyExampleTable].[ID] ASC,?=@HandlingStatus1 tinyint,@HandlingStatus1=0)'
TDS: head_len=8, payload=182, type=SQL batch, status=0x1
TDS(msg): head_len=182, payload=0, Clt->Srv, query, query=' set transaction isolation level  read committed  set implicit_transactions off '
TDS: head_len=8, payload=26, type=Tabular result, status=0x1
TDS(msg): head_len=26, payload=0, Srv->Clt, query, nb_rows=-1, nb_fields=-1, Status=Completed
TDS: head_len=8, payload=245, type=RPC, status=0x1
TDS(msg): head_len=245, payload=0, Clt->Srv, query, query='PrepExec(?*=0,?=NULL,?=create table newsyb (column1 char(30) not null, column2 char(30) null,column3 char(30) null))'
TDS: head_len=8, payload=49, type=Tabular result, status=0x1
TDS(msg): head_len=49, payload=0, Srv->Clt, query, nb_rows=-1, nb_fields=-1, Status=Completed
TDS: head_len=8, payload=225, type=RPC, status=0x1
TDS(msg): head_len=225, payload=0, Clt->Srv, query, query='PrepExec(?*=0,?=NULL,?=insert INTO newsyb (column1, column2, column3) VALUES ('first', 'second', 'third'))'
TDS: head_len=8, payload=49, type=Tabular result, status=0x1
TDS(msg): head_len=49, payload=0, Srv->Clt, query, nb_rows=1, nb_fields=-1, Status=Completed
TDS: head_len=8, payload=53, type=RPC, status=0x1
TDS(msg): head_len=53, payload=0, Clt->Srv, query, query='Execute(?=2)'
TDS: head_len=8, payload=72, type=Tabular result, status=0x1
TDS(msg): head_len=72, payload=0, Srv->Clt, query, nb_rows=1, nb_fields=-1, Status=Completed
TDS: head_len=8, payload=101, type=RPC, status=0x1
TDS(msg): head_len=101, payload=0, Clt->Srv, query, query='PrepExec(?*=0,?=NULL,?=select * from newsyb)'
TDS: head_len=8, payload=430, type=Tabular result, status=0x1
TDS(msg): head_len=430, payload=0, Srv->Clt, query, nb_rows=3, nb_fields=3, Status=Completed
TDS: head_len=8, payload=37, type=RPC, status=0x1
TDS(msg): head_len=37, payload=0, Clt->Srv, query, query='Execute(?=3)'
TDS: head_len=8, payload=412, type=Tabular result, status=0x1
TDS(msg): head_len=412, payload=0, Srv->Clt, query, nb_rows=3, nb_fields=3, Status=Completed
TDS: head_len=8, payload=95, type=RPC, status=0x1
TDS(msg): head_len=95, payload=0, Clt->Srv, query, query='PrepExec(?*=0,?=NULL,?=drop table newsyb)'
TDS: head_len=8, payload=49, type=Tabular result, status=0x1
TDS(msg): head_len=49, payload=0, Srv->Clt, query, nb_rows=-1, nb_fields=-1, Status=Completed
TDS: head_len=8, payload=7992, type=RPC, status=0x4
TDS: head_len=8, payload=331, type=RPC, status=0x1
TDS(msg): head_len=8323, payload=0, Clt->Srv, query, query='p_SaveExample(@LongParam=Studenckie Koło Przewodników Turystycznych w Gdańsku
TDS: head_len=8, payload=363, type=RPC, status=0x9
TDS(msg): head_len=363, payload=0, Clt->Srv, query, query='p_SetBogusSample(@BogusDetailsID=74565,@BogusStatusID=5,@ResultCode=NULL,@ResultMsg=NULL,@ErrorCode=NULL,@ErrorMsg=NULL,@ExampleBogusGeneratedID=NULL,@ExampleType=1)'
TDS: head_len=8, payload=80, type=RPC, status=0x9
TDS(msg): head_len=80, payload=0, Clt->Srv, query, query='p_GetMyExampleTableRowCount()'
TDS: head_len=8, payload=210, type=RPC, status=0x1
TDS(msg): head_len=210, payload=0, Clt->Srv, query, query='proc_GetMyExampleTableSampleMetaData(?=33221100554477668899aabbccddeeff,?=NULL,?=NULL,?=ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghij,?=1,?=45,?=0123456789abcdeffedcba98,?=108)'
TDS: head_len=8, payload=191, type=RPC, status=0x9
TDS(msg): head_len=191, payload=0, Clt->Srv, query, query='proc_GetMyExampleTableSampleMetaData(?=00112233445566778899aabbccddeeff,?=NULL,?=Bogus,?=NULL,?=1,?=45,?=0123456789abcdefedcba9876543210123456789abcdefedcba98765,?=18)'
TDS: head_len=8, payload=260, type=RPC, status=0x9
TDS(msg): head_len=260, payload=0, Clt->Srv, query, query='proc_FetchMyExampleData(?=0123456789abcdef0123456789abcdef,?=NULL,?=BOGUS,?=0,?=4294967294,?=0,?=0,?=NULL,?=NULL,?=NULL,?=0,?=0123456789abcdef0123456789abcdef0123456789abcdef01234567,?=NULL,?=1,?=5242880,?=45,?=1,?=0,?=NULL,?*=NULL)'
TDS: head_len=8, payload=312, type=RPC, status=0x9
TDS(msg): head_len=312, payload=0, Clt->Srv, query, query='dbo.proc_GetMySampleDataItems(@SampleItemId=661ac34e14a25348a77ee7060fffff07,@DataId=NULL,@DataItemType=9b45b9bca8836445b1d3e9e198478f4e,@TableId=NULL,@MaxFetchSize=100,@SomeOtherSampleId=0)'
-- pcap/tds/mssql_truncated.pcap --
TDS: head_len=8, payload=59, type=Pre-login, status=0x1
TDS(msg): head_len=59, payload=0, Clt->Srv, version=0.0, startup, No SSL
TDS: head_len=8, payload=23, type=Tabular result, status=0x1
TDS(msg): head_len=23, payload=0, Srv->Clt, version=11.0, startup, No SSL
TDS: head_len=8, payload=154, type=Pre-login, status=0x1
TDS(msg): head_len=154, payload=0, Clt->Srv, startup, No SSL
TDS: head_len=8, payload=609, type=Pre-login, status=0x1
TDS(msg): head_len=609, payload=0, Srv->Clt, startup, No SSL
TDS: head_len=8, payload=198, type=Pre-login, status=0x1
TDS(msg): head_len=198, payload=0, Clt->Srv, startup, No SSL
TDS: head_len=8, payload=59, type=Pre-login, status=0x1
TDS(msg): head_len=59, payload=0, Srv->Clt, startup, No SSL
TDS: head_len=8, payload=409, type=Tabular result, status=0x1
TDS(msg): head_len=409, payload=0, Srv->Clt, version=7.4, startup, No SSL, dbname=octo, Status=Completed
TDS: head_len=8, payload=99, type=RPC, status=0x1
TDS(msg): head_len=99, payload=0, Clt->Srv, query, query='PrepExec(?*=0,?=NULL,?=SELECT * FROM Toto )'
TDS: head_len=8, payload=7992, type=Tabular result, status=0x0
TDS(msg): head_len=2888, payload=0, Srv->Clt, query, nb_rows=57, nb_fields=2
TDS: head_len=8, payload=7992, type=Tabular result, status=0x0
TDS(msg): head_len=7992, payload=0, Srv->Clt, query, nb_rows=-1, nb_fields=-1
TDS: head_len=8, payload=7992, type=Tabular result, status=0x0
TDS(msg): head_len=7992, payload=0, Srv->Clt, query, nb_rows=-1, nb_fields=-1
TDS: head_len=8, payload=7992, type=Tabular result, status=0x0
TDS(msg): head_len=7992, payload=0, Srv->Clt, query, nb_rows=-1, nb_fields=-1
TDS: head_len=8, payload=7992, type=Tabular result, status=0x0
TDS(msg): head_len=18816, payload=0, Srv->Clt, query, nb_rows=-1, nb_fields=-1
-- pcap/tds/mssql_with_accent.pcap --
TDS: head_len=8, payload=59, type=Pre-login, status=0x1
TDS(msg): head_len=59, payload=0, Clt->Srv, version=0.0, startup, No SSL
TDS: head_len=8, payload=23, type=Tabular result, status=0x1
TDS(msg): head_len=23, payload=0, Srv->Clt, version=11.0, startup, No SSL
TDS: head_len=8, payload=154, type=Pre-login, status=0x1
TDS(msg): head_len=154, payload=0, Clt->Srv, startup, No SSL
TDS: head_len=8, payload=609, type=Pre-login, status=0x1
TDS(msg): head_len=609, payload=0, Srv->Clt, startup, No SSL
TDS: head_len=8, payload=198, type=Pre-login, status=0x1
TDS(msg): head_len=198, payload=0, Clt->Srv, startup, No SSL
TDS: head_len=8, payload=59, type=Pre-login, status=0x1
TDS(msg): head_len=59, payload=0, Srv->Clt, startup, No SSL
TDS: head_len=8, payload=429, type=Tabular result, status=0x1
TDS(msg): head_len=429, payload=0, Srv->Clt, version=7.4, startup, No SSL, dbname=ç é ààààà, Status=Completed
TDS: head_len=8, payload=227, type=RPC, status=0x1
TDS(msg): head_len=227, payload=0, Clt->Srv, query, query='PrepExec(?*=0,?=@P0 nvarchar(4000),?=DELETE FROM Toto WHERE name = @P0ç é ààààà        ,?=ç é ààààà)'
TDS: head_len=8, payload=288, type=Tabular result, status=0x1
TDS(msg): head_len=288, payload=0, Srv->Clt, query, nb_rows=-1, nb_fields=-1, Status=Error, ErrorCode=137, ErrorMessage=Must declare the scalar variable "@P0ç".
